src/calibrationlib/calibration_benchmark_runner.cpp

```diff
@@ -0,0 +1,104 @@
+#include "calibration_benchmark_runner.hpp"
+#include <benchmark_runner.hpp>
+#include <file_based_benchmark_item_runner.hpp>
+#include <file_based_table_generator.hpp>
+#include <tpcds/tpcds_table_generator.hpp>
+#include <tpch/tpch_benchmark_item_runner.hpp>
+#include <tpch/tpch_table_generator.hpp>
+#include "hyrise.hpp"
+
+namespace opossum {
+
+CalibrationBenchmarkRunner::CalibrationBenchmarkRunner(const std::string& path_to_dir)
+    : _feature_export(OperatorFeatureExport(path_to_dir)), _table_export(TableFeatureExport(path_to_dir)) {
+  auto config = std::make_shared<BenchmarkConfig>(BenchmarkConfig::get_default_config());
+  config->max_runs = 1;
+  config->enable_visualization = false;
+  config->chunk_size = 100'000;
+  config->cache_binary_tables = true;
+  _config = config;
+}
+
+CalibrationBenchmarkRunner::CalibrationBenchmarkRunner(const std::string& path_to_dir,
+                                                       std::shared_ptr<BenchmarkConfig> config)
+    : _feature_export(OperatorFeatureExport(path_to_dir)), _table_export(TableFeatureExport(path_to_dir)) {
+  _config = config;
+}
+
+void CalibrationBenchmarkRunner::run_benchmark(const BenchmarkType type, const float SCALE_FACTOR,
+                                               const int number_of_executions) {
+  std::shared_ptr<BenchmarkRunner> benchmark_runner;
+  switch (type) {
+    case BenchmarkType::TCPH:
+      benchmark_runner = _build_tcph(SCALE_FACTOR);
+      break;
+    case BenchmarkType::TCPDS:
+      benchmark_runner = _build_tcpds(SCALE_FACTOR);
+      break;
+    case BenchmarkType::JOB:
+      benchmark_runner = _build_job();
+      break;
+    default:
+      throw std::runtime_error("Provided unknown BenchmarkType.");
+  }
+
+  for (int i = 0; i < number_of_executions; ++i) {
+    Hyrise::get().benchmark_runner = benchmark_runner;
+    benchmark_runner->run();
+
+    auto& pqp_cache = Hyrise::get().default_pqp_cache;
+
+    for (auto pqp_entry = pqp_cache->unsafe_begin(); pqp_entry != pqp_cache->unsafe_end(); ++pqp_entry) {
+      const auto& [query_string, physical_query_plan] = *pqp_entry;
+      _feature_export.export_to_csv(physical_query_plan);
+    }
+
+    // Clear pqp cache for next benchmark
+    pqp_cache->clear();
+  }
+
+  const std::vector<std::string> table_names = Hyrise::get().storage_manager.table_names();
+  for (const auto& table_name : table_names) {
+    auto table = Hyrise::get().storage_manager.get_table(table_name);
+    _table_export.export_table(std::make_shared<CalibrationTableWrapper>(CalibrationTableWrapper(table, table_name)));
+
+    // Drop table because we do not need it anymore
+    Hyrise::get().storage_manager.drop_table(table_name);
```

Comment:
Usually, we don't drop tables for benchmarks because when the binary exits, everything is "dropped". However, it does not harm and there might be cases where we run multiple benchmarks with different scale factors after each other for calibration.

But I think the comment is unnecessary.